tag_updater.py

```diff
@@ -0,0 +1,72 @@
+import json
+import planning_domains_api as api
+import ast
+
+# Pull requirments from precomputed json file
+f = open("processed_result5.json")
+processed_requirements = json.load(f)
+
+DEBUG = True
+
+def update_tags(resource, id, *, current, required):
+    assert resource in ["collection", "domain", "problem"]   
+
+    # Remove incorrect tags
+    for tag in current:
+        if tag not in required:
+            if DEBUG:
+                print(f"Untagging {tag} from {resource}: {id}")
+            elif resource == "collection":
+                api.untag_collection(id, tag)
+            elif resource == "domain":
+                api.untag_domain(id, tag)
+            elif resource == "problem":
+                api.untag_problem(id, tag)
+
+    # Add required tags
+    for tag in required:
+        if tag not in current:
+            if DEBUG:
+                print(f"Tagging {tag} from {resource}: {id}")
+            elif resource == "collection":
+                api.tag_collection(id, tag)
+            elif resource == "domain":
+                api.tag_domain(id, tag)
+            elif resource == "problem":
+                api.tag_problem(id, tag)
+
+collections = api.get_collections()
+for collection in collections:
+    collection_id = int(collection["collection_id"])
+
+    # Keep track of all requirements inside this collection
+    collection_required_tags = set()
+
+    domains = api.get_domains(collection_id)
+    assert ast.literal_eval(collection['domain_set']).sort() == [x['domain_id'] for x in domains].sort(), "domain_set property should contain all domains"
+    for domain in domains:
+        domain_id = domain['domain_id']
+
+        if domain_id == 80: # This domain 404's could not read parse domain reqs
+            continue
+        if domain_id == 21: # This domain is the settlers domain, and is not classical
+            continue
+        if domain_id == 145 or domain_id == 133: # These domains cannot be parsed with VAl because of the "?after" parsing bug
+            continue
```

Comment:
Couple of edge cases. I was thinking we could update these tags manually

Reply:
@Rowan441 : Can you put in a PR for the classical benchmark repo to replace `?after` with something that VAL is ok with?

Reply:
Fixed the domain_ids:
 - 21: removed since this no longer exists
 - 80: ran Parser on files from classical-domains repo and added to processed results:
```json
"80": {
    "val": [
      ":action-costs"
    ]
  }
```
 - 145 & 133: ran Parser on files from classical-domains repo with `?after` changed and added to processed results:
```json
"145": {
      "val": [
        ":typing",
        ":negative-preconditions",
        ":action-costs"
      ]
  },
...
  "133": {
      "val": [
        ":typing",
        ":negative-preconditions",
        ":action-costs"
      ]
  },
```